src/python/pybind11/src/pyxrt.cpp

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright (C) 2016-2020 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include <cstring>
+#include <tuple>
+#include <typeinfo>
+
+// XRT includes
+#include "experimental/xrt_device.h"
+#include "experimental/xrt_kernel.h"
+#include "experimental/xrt_bo.h"
+#include "xrt.h"
+
+// Pybind11 includes
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/numpy.h>
+
+namespace py = pybind11;
+
+PYBIND11_MODULE(pyxrt, m) {
+m.doc() = "Pybind11 module for XRT";
+
+/*
+ *
+ * Constants and Enums
+ *
+ *
+ */
+m.attr("XCL_BO_FLAGS_NONE") = py::int_(XCL_BO_FLAGS_NONE);
+
+py::enum_<xclBOSyncDirection>(m, "xclBOSyncDirection")
+    .value("XCL_BO_SYNC_BO_TO_DEVICE", xclBOSyncDirection::XCL_BO_SYNC_BO_TO_DEVICE)
+    .value("XCL_BO_SYNC_BO_FROM_DEVICE", xclBOSyncDirection::XCL_BO_SYNC_BO_FROM_DEVICE);
+
+
+/*
+ *
+ * XRT:: Functions
+ *
+ */
+m.def("xclProbe", &xclProbe); 
```

Comment:
Don't wrap this function.   It's only purpose is to provide the number of boards hooked up to the system.  You don't need it, since opening of the device will (gracefully) fail if you provide a device index that doesn't exists.   I suspect xclProbe is the only reason you link with xrt_core.  If you check all the low level tests (tests/xrt) you will see I recently  removed the call to xclProbe from all of these.

Reply:
removed xclprobe from cpp and README ... calling device(0) passing simple test